### DISCUSSION TITLE
Checkstyle

### DISCUSSION THREAD

**Discussion by brenuart** [logfellow/logstash-logback-encoder#478] [Ideas]
Should we add some checks during the build to enforce some coding conventions?
I'm thinking about SPACES instead of TABS for instances... (because I'm pretty confident I will forget once again to configure my IDE accordingly)...

Something like maven-checkstyle-plugin with a very limited set of checks may be ?
Are you aware of any other tool ?

**Comment by philsttr** [comment]
Yeah, I'll work on that.

**Reply by brenuart** [reply]
Check this for a first quick setup: https://github.com/brenuart/logstash-logback-encoder/tree/checkstyle

**Comment by philsttr** [comment]
Fixed in https://github.com/logstash/logstash-logback-encoder/commit/bb42b6b9bbd3e9843a133cc0291d15ea21b20e98

I'll convert this discussion to an issue after github re-enables that functionality.  Converting between issues <-> discussions is currently disabled due to some bugs.

**Reply by philsttr** [reply]
Apparently github only supports moving issues -> discussion.  Not from discussion -> issue.

**Comment by brenuart** [comment]
Made my first PR since long - first time I see the check style rules in action ;-)
A first comment about the _RightCurly_ rule. This rule complains about the following code:

```
if (condition == true) {
   ... do something ...
}
else {
   ... do something else...
}
```

According to the rule this should be written as follows:
```
if (condition == true) {
   ... do something ...
} else {
   ... do something else...
}
```

Personally, I prefer the first option as it clearly separates the _if_ and _else_ blocks thanks to the extra (nearly) blank line.
Same for a _catch_ block:
```
try {
   ... do something ...
}
catch(Exception e) {
  ... do something else ...
}
finally {
  ...
}
```
will be refused and should instead be written as follows:
```
try {
   ... do something ...
} catch(IOException e) {
  ... do something else ...
} finally {
 ...
}
```
Again, I prefer the first approach for the same reasons as above.

All of this is of course a matter of taste... but since the introduction of checkstyle is rather new, I was wondering if these rules were added "on purpose" or "by default". So my feedback...

**Reply by philsttr** [reply]
I created the rules to match the majority practice used in the existing codebase, so the entire codebase did not need to be reformatted.  I know formatting is very opinionated, but I'd like to not reformat the existing code.

**Reply by brenuart** [reply]
Ok.. I'll get use to it. Just a matter of finding how to configuré Eclipse to help me with this.

**Comment by jug** [comment]
could also use `.editorconfig` to format the code already during editing for IDEs
